code/modules/antagonists/cult/cult_items.dm

```diff
@@ -875,26 +875,26 @@
 	ADD_TRAIT(src, TRAIT_NODROP, CULT_TRAIT)
 
 
-/obj/item/blood_beam/afterattack(atom/A, mob/living/user, flag, params)
+/obj/item/blood_beam/afterattack(atom/A, mob/living/user, proximity_flag, clickparams)
 	. = ..()
 	if(firing || charging)
 		return
-	var/C = user.client
-	if(ishuman(user) && C)
-		var/list/angle_vector = calculate_projectile_angle_and_pixel_offsets(user, params)
-		angle = angle_vector[1]
+	if(ishuman(user))
+		angle = Get_Angle(user, A)
 	else
 		qdel(src)
 		return
 	charging = TRUE
 	INVOKE_ASYNC(src, .proc/charge, user)
-	if(do_after(user, 90, target = user))
+	if(do_after(user, 9 SECONDS, target = user))
 		firing = TRUE
-		INVOKE_ASYNC(src, .proc/pewpew, user, params)
+		ADD_TRAIT(user, TRAIT_IMMOBILIZED, CULT_TRAIT)
```

Comment:
I'm not a fan of the immobilization here- it's almost certainly to cause another bug at some point, and you can still cheese the spell by having a shade or something stand next to you and move you near the end of the beam's duration.

honestly, it would probably be an improvement if we just removed the 4 second hardstun for letting the beam go for its full duration. blood beam is absurdly impractical anyway; we don't need to punish people for having fun with their doom laser for too long.